src/cryptography/x509/ocsp.py

```diff
@@ -98,6 +106,159 @@ def build(self):
+                "algorithm must be SHA1, SHA224, SHA256, SHA384, or SHA512"
+            )
+        if not isinstance(this_update, datetime.datetime):
+            raise TypeError("this_update must be a datetime object")
+        if (
+            next_update is not None and
+            not isinstance(next_update, datetime.datetime)
+        ):
+            raise TypeError("next_update must be a datetime object or None")
+
+        self._cert = cert
+        self._issuer = issuer
+        self._algorithm = algorithm
+        self._this_update = this_update
+        self._next_update = next_update
+
+    def set_status(self, cert_status, revocation_time, revocation_reason):
+        if not isinstance(cert_status, OCSPCertStatus):
+            raise TypeError(
+                "cert_status must be an item from the OCSPCertStatus enum"
+            )
+        if cert_status is not OCSPCertStatus.REVOKED:
+            if revocation_time is not None:
+                raise ValueError(
+                    "revocation_time can only be provided if the certificate "
+                    "is revoked"
+                )
+            if revocation_reason is not None:
+                raise ValueError(
+                    "revocation_reason can only be provided if the certificate"
+                    " is revoked"
+                )
+        else:
+            if not isinstance(revocation_time, datetime.datetime):
+                raise TypeError("revocation_time must be a datetime object")
+
+            revocation_time = _convert_to_naive_utc_time(revocation_time)
+            if revocation_time <= _UNIX_EPOCH:
+                raise ValueError('The revocation_time must be after the unix'
+                                 ' epoch (1970 January 1).')
+
+            if (
+                revocation_reason is not None and
+                not isinstance(revocation_reason, x509.ReasonFlags)
+            ):
+                raise TypeError(
+                    "revocation_reason must be an item from the ReasonFlags "
+                    "enum or None"
+                )
+
+        self._cert_status = cert_status
+        self._revocation_time = revocation_time
+        self._revocation_reason = revocation_reason
+
+
+class OCSPResponseBuilder(object):
+    def __init__(self, response=None, responder_id=None, certs=None,
+                 extensions=[]):
+        self._response = response
+        self._responder_id = responder_id
+        self._certs = certs
+        self._extensions = extensions
+
+    def add_response(self, cert, issuer, algorithm, cert_status, this_update,
+                     next_update, revocation_time, revocation_reason):
+        if self._response is not None:
+            raise ValueError("Only one response per OCSPResponse.")
+
+        singleresp = _SingleResponse(
+            cert, issuer, algorithm, this_update, next_update
+        )
+        singleresp.set_status(cert_status, revocation_time, revocation_reason)
```

Comment:
The only reason was that `__init__` becomes very large and the constructor takes 8 arguments at that point. Since it's purely private API I'm fine either way though. Want me to change it?

Reply:
Yes please :-)